Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using UnityEngine;
+#if UNITY_2021_1_OR_NEWER
+using UnityEngine.Pool;
+#else
+using UnityEngine.SceneManagement;
+using System.Collections.Generic;
+#endif
+
+namespace Perigon.Utility
+{
+    public class ObjectPooler<T> where T : Component
+    {
+		protected string name;
+
+#if UNITY_2021_1_OR_NEWER
+		protected IObjectPool<T> objectPool;
+#else
+		protected Scene poolScene;
+		protected List<T> pool = null;
+#endif
+
+		protected Func<T> createFunc;
+		protected Action<T> actionOnGet;
+		protected Action<T> actionOnRelease;
+
+        public ObjectPooler(string poolName, Func<T> createFunc, Action<T> actionOnGet, Action<T> actionOnRelease)
+        {
+            this.name = poolName;
+            this.createFunc = createFunc;
+            this.actionOnGet = actionOnGet;
+            this.actionOnRelease = actionOnRelease;
+
+#if UNITY_2021_1_OR_NEWER
+			objectPool = new ObjectPool<T>(createFunc, actionOnGet, actionOnRelease);
+#endif
+		}
+
+        public T Get()
+		{
+
+#if UNITY_2021_1_OR_NEWER
+			return objectPool.Get();
+#else
+			T instance;
+			if (pool == null)
+			{
+				CreatePools();
+			}
+			int lastIndex = pool.Count - 1;
+			if (lastIndex >= 0)
+			{
+				instance = pool[lastIndex];
+				pool.RemoveAt(lastIndex);
+			}
+			else
+			{
+				instance = createFunc();
+				SceneManager.MoveGameObjectToScene(
+					instance.gameObject, poolScene
+				);
+			}
+			actionOnGet?.Invoke(instance);
+			return instance;
+#endif
+		}
+
+		public void Reclaim(T toRecycle)
+		{
+#if UNITY_2021_1_OR_NEWER
+			objectPool.Release(toRecycle);
+#else
+			if (pool == null)
+			{
+				CreatePools();
+			}
+			actionOnRelease?.Invoke(toRecycle);
+			pool.Add(toRecycle);
+#endif
+		}
+
+#if !UNITY_2021_1_OR_NEWER
```

Comment:
why do you need this here if it's going to compile anyway?

Reply:
I mean its not strictly necessary, but just to keep it simpler in Unity 2021 I wrapped all the unneeded scene management stuff in conditional flags. This code should never run in 2021, so I figure it makes sense to just keep it out

Reply:
That's fine for now, just can be a bit confusing readability wise.

Reply:
Yeah tabs and color formatting are doing a lot of work, I'm not sure if you're just reading it on github but it is a bit better off of github